File: src/main/java/com/example/demo/model/ReportJob.java
package com.example.demo.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.time.OffsetDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ReportJob {
    private Long id;
    private String username;
    private LocalDate fromDate;
    private LocalDate toDate;
    private String format;
    @Builder.Default
    private String status = "PENDING";
    private String filePath;
    private String errorMessage;
    private OffsetDateTime createdAt;
    private OffsetDateTime updatedAt;
}

File: src/main/kotlin/com/example/demo/model/ReportJob.kt
package com.example.demo.model

import java.time.OffsetDateTime

class ReportJob(
    var id: Long? = null,
    var username: String,
    var fromDate: java.time.LocalDate,
    var toDate: java.time.LocalDate,
    var format: String,
    var status: String = "PENDING",
    var filePath: String? = null,
    var errorMessage: String? = null,
    var createdAt: OffsetDateTime? = null,
    var updatedAt: OffsetDateTime? = null
)

